vpr/src/base/vpr_context.h

```diff
@@ -520,6 +520,17 @@ struct FloorplanningContext : public Context {
      */
     vtr::vector<ClusterBlockId, PartitionRegion> cluster_constraints;
 
+    /**
+     * @brief Floorplanning constraints in the compressed grid coordinate system.
+     *
+     * Each clustered block has a logical type with a corresponding compressed grid.
+     * Compressed floorplanning constraints are calculated by translating the grid locations
+     * of floorplanning regions to compressed grid locations. To ensure regions do not enlarge:
+     * - The bottom left corner is rounded up to the nearest compressed location.
+     * - The top right corner is rounded down to the nearest compressed location.
+     */
```

Comment:
Update the comment to say how this works.  Also, should we make the first index the ClusterBlockId (should be slightly faster for 3D, multi-layer case, although for 2D case having layer be the first index could be faster)?
Also, should we just do this intersection in normal grid space?
Comment indices [0..grid.num_layers-1][0..numClusters-1]